components/datadog/agent/kubernetes_helm.go

```diff
@@ -361,6 +364,17 @@ func (values HelmValues) configureImagePullSecret(secret *corev1.Secret) {
 	}
 }
 
+func (values HelmValues) configureEnvVars(envVars map[string]string) {
```

Comment:
If you are talking about providing the `envDict` through the `ValuesYAML` parameter, I tried but this is not working. Because if you use a fakeintake, [here](https://github.com/DataDog/test-infra-definitions/blob/main/components/datadog/agent/kubernetes_helm.go#L132) we modify `values` with the environment variables needed to configure the fakeintake. When we pass `ValuesYAML` and `values` to `helm.NewInstallation` the env variables you define in `ValuesYAML` are not merged with the one in `values`

Reply:
I meant using using the `envDict` variable in the Helm chart: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L531-L533.

The merge issues arises because `lists` merge is override, it's the behaviour in Helm (which is not used in Pulumi but replicated [here](https://github.com/pulumi/pulumi-kubernetes/blob/78ab38748b9101d98890d86af44f5addb6c852dd/provider/pkg/provider/helm_release.go#L1323))

Using `envDict` allow to define env vars in a map, which support merging, with `Values` overriding `ValuesYAML` if duplicate keys. If it's not working, there's something to be investigated.

Reply:
Oh yes, I am not very familiar with the Agent helm chart but that looks like something I could use! I will try to confirm it works!

Reply:
Removed these as it is not needed with `envDict`